presto-main/src/main/java/com/facebook/presto/sql/planner/plan/TableScanNode.java

```diff
@@ -81,9 +81,21 @@ public TableScanNode(
         this.table = table;
         this.outputSymbols = ImmutableList.copyOf(outputs);
         this.assignments = ImmutableMap.copyOf(assignments);
-        this.originalConstraint = originalConstraint;
         this.tableLayout = tableLayout;
         this.currentConstraint = currentConstraint;
+        this.originalConstraint = originalConstraint;
+    }
+
+    @JsonCreator
```

Comment:
Can you move this version of the constructor above the other one? We normally delegate down, and moving it should minimize the diff.